R/misc.R

```diff
@@ -21,13 +21,21 @@ get_rhs_vars <- function(formula, data, no_lhs = FALSE) {
   ## or should it? what about Y ~ log(x)?
   ## Answer: when called from `form2args`, the function
   ## `inline_check` stops when in-line functions are used.
-  data_info <- attr(model.frame(formula, data[1, ]), "terms")
-  response_info <- attr(data_info, "response")
-  predictor_names <- names(attr(data_info, "dataClasses"))
-  if (length(response_info) > 0 && all(response_info > 0)) {
-    predictor_names <- predictor_names[-response_info]
+
+  outcomes_names <- all.names(rlang::f_lhs(formula), functions = FALSE)
+
+  formula_rhs <- rlang::f_rhs(formula)
+  if (identical(formula_rhs, quote(.))) {
```

Comment:
I believe this is what you're getting at in your comment above, but confirming that this doesn't need it's own `all.names()` magic to accommodate formulas that use . in tricky ways? I see e.g. `(.)^2` or `log(.)` is handled by the above.

Reply:
so `(.)^2` and `log(.)` wont happen, because `^` and `log()` function calls that will be errored on earlier in the code.

however we need special handling of `.` because it is special

``` r
formula <- mpg ~ .

rlang::f_rhs(formula)
#> .
all.names(rlang::f_rhs(formula))
#> [1] "."
```

Reply:
but I just noticed that the above only works if `.` is alone on the RHS, i'll fix before requesting review

``` r
library(recipes)

recipe(mpg ~ ., mtcars)
#> 
#> ── Recipe ──────────────────────────────────────────────────────────────────────
#> 
#> ── Inputs
#> Number of variables by role
#> outcome:    1
#> predictor: 10

recipe(mpg ~ . + disp, mtcars)
#> Error in `data[, vars]` at recipes/R/recipe.R:252:3:
#> ! Can't subset columns that don't exist.
#> ✖ Column `.` doesn't exist.
```